packages/realm-server/tests/realm-endpoints/has-private-dependencies-test.ts

```diff
@@ -0,0 +1,707 @@
+      [],
+      'No violations reported',
+    );
+  });
+
+  test('lists direct dependencies on private realms', async function (assert) {
+    let { url: privateRealmURL } = await createRealm({
+      name: 'Private Realm',
+      files: {
+        'secret-card.gts': `
+          import { contains, field, CardDef } from "https://cardstack.com/base/card-api";
+          import StringField from "https://cardstack.com/base/string";
+
+          export class SecretCard extends CardDef {
+            @field name = contains(StringField);
+          }
+        `,
+      },
+    });
+
+    let { url: sourceRealmURL, realm: sourceRealm } = await createRealm({
+      name: 'Source Realm',
+      files: {
+        'source-card.gts': `
+          import {
+            contains,
+            field,
+            linksTo,
+            CardDef,
+          } from "https://cardstack.com/base/card-api";
+          import StringField from "https://cardstack.com/base/string";
+          import { SecretCard } from "${privateRealmURL}secret-card";
+
+          export class SourceCard extends CardDef {
+            @field label = contains(StringField);
+            @field secret = linksTo(() => SecretCard);
+          }
+        `,
+        'source-instance.json': {
+          data: {
+            type: 'card',
+            attributes: {
+              label: 'Secret label',
+            },
+            meta: {
+              adoptsFrom: {
+                module: './source-card',
+                name: 'SourceCard',
+              },
+            },
+          },
+        },
+      },
+    });
+
+    let response = await request
+      .get(`${new URL(sourceRealmURL).pathname}_has-private-dependencies`)
+      .set('Accept', SupportedMimeType.JSONAPI)
+      .set(
+        'Authorization',
+        `Bearer ${createJWT(sourceRealm, ownerUserId, DEFAULT_PERMISSIONS)}`,
+      );
+
+    assert.strictEqual(response.status, 200, 'HTTP 200 status');
+    assert.false(
+      response.body.data.attributes.publishable,
+      'Realm is not publishable',
+    );
+
+    assert.strictEqual(
+      response.body.data.attributes.violations.length,
+      1,
+      'one violating resource is reported',
+    );
+    let violation = response.body.data.attributes.violations[0];
+    assert.strictEqual(
+      violation.resource,
+      `${sourceRealmURL}source-instance.json`,
+      'violation references the offending instance',
+    );
+    assert.true(
+      violation.externalDependencies.some((dep: any) =>
+        String(dep.dependency).startsWith(`${privateRealmURL}secret-card`),
+      ),
+      'dependency points into the private realm',
+    );
+    assert.true(
+      violation.externalDependencies.every(
+        (dep: any) => dep.realmURL === privateRealmURL,
+      ),
+      'dependencies report the private realm URL',
+    );
+  });
+
+  test('traces transitive dependencies through public realms', async function (assert) {
```

Comment:
I'm not convinced we need to do this. Stopping at a dep in a public realm seems like a pretty darn good optimization.

Reply:
ya… I had some convoluted scenario in mind but it really doesn’t make sense, at least with the current setup, so removed now, thanks